### PR TITLE
[IOTDB-4334] Verify durable cases of RegionGroup extension

### PR DESCRIPTION
**Pull request by CRZbulabula**
**Algorithm:**

When received new Partitions creation requests and there are not available RegionGroups in the corresponding StorageGroup, we should **create a new RegionGroup** to ensure the new Partition can be created successfully.

Since the related function is completed in previous PRs, I just enhance the IoTDBPartitionDurableIT in this PR. You can see the [RegionGroup allocation algorithm](https://apache-iotdb.feishu.cn/docs/doccnHuDkDD6RiwLjaej3hkA9Vb) doc to get more details of relevant algorithm.

**Updates:**

1. testRemovingDataNode: set a DataNode to Removing status, the newborn RegionGroup should keep away from the Removing one.
2. testReadOnlyDataNode: set a DataNode to ReadOnly status, the newborn RegionGroup should keep away from the ReadOnly one.
3. testUnknownDataNode: shutdown a DataNode, the RegionGroup should still create successfully, and the RegionMaintainer can re-create the releated Regions.

### CODE REVIEW

File: integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionDurableIT.java [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+        TimeUnit.SECONDS.sleep(1);
+      }
+
+      /* Test getOrCreateSchemaPartition, the result should be NO_ENOUGH_DATANODE */
+      schemaPartitionReq =
+          new TSchemaPartitionReq()
+              .setPathPatternTree(ConfigNodeTestUtils.generatePatternTreeBuffer(new String[] {d1}));
+      schemaPartitionTableResp = client.getOrCreateSchemaPartitionTable(schemaPartitionReq);
+      Assert.assertEquals(
+          TSStatusCode.NO_AVAILABLE_REGION_GROUP.getStatusCode(),
```
Comment (Caideyipi) [inline integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionDurableIT.java:229]:
```suggestion
          TSStatusCode.NO_ENOUGH_DATANODE.getStatusCode(),
```

**Review by Caideyipi — COMMENTED** [review]
PTAL~

File: integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionDurableIT.java [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+                  }
+                });
+
+        if (containRemoving.get()) {
+          break;
+        }
+        TimeUnit.SECONDS.sleep(1);
+      }
+
+      /* Test getOrCreateSchemaPartition, the result should be NO_ENOUGH_DATANODE */
```
Comment (Caideyipi) [inline integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionDurableIT.java:223]:
```suggestion
      /* Test getOrCreateSchemaPartition, the result should be NO_AVAILABLE_REGION_GROUP */
```

Comment (CRZbulabula) [reply integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionDurableIT.java]: Fixed

**Review by Beyyes — APPROVED** [review]

File: integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionGetterIT.java [hunk 3]
```diff
@@ -94,9 +96,9 @@ public class IoTDBPartitionGetterIT {
 
   private static final String sg = "root.sg";
   private static final int storageGroupNum = 5;
-  private static final int seriesPartitionSlotsNum = 10000;
-  private static final int seriesPartitionBatchSize = 100;
-  private static final int timePartitionSlotsNum = 10;
+  private static final int testSeriesPartitionSlotNum = 100;
```
Comment (Beyyes) [inline integration-test/src/test/java/org/apache/iotdb/confignode/it/partition/IoTDBPartitionGetterIT.java:99]: This test fails twice

File: integration-test/src/main/java/org/apache/iotdb/itbase/env/BaseEnv.java [hunk 4]
```diff
@@ -189,12 +189,22 @@ default SessionPool getSessionPool(
...
+  /** @return The TDataNodeLocation of the specified DataNode */
+  DataNodeWrapper getDataNodeWrapper(int index);
+
+  /** Register a new DataNode */
+  void registerNewDataNode();
+
+  /** Start an existed DataNode */
   void startDataNode(int index);
 
+  /** Start an existed DataNode */
```
[the remark this answers is not in the document]
Comment (CRZbulabula) [reply integration-test/src/main/java/org/apache/iotdb/itbase/env/BaseEnv.java:207]: FIxed~

**Pull request merged**